### DISCUSSION TITLE
V2 Programming Model for Python

### DISCUSSION THREAD

**Discussion by shreyabatra4** [Azure/azure-functions-python-worker#1381] [General]
Let us know any feedback and feature requests for the v2 programming model for Python!

**Comment by LouisDeconinck** [comment]
Is it possible to create a binding between Azure Function Python V2 and Azure SQL Database? If so, how?

**Reply by shreyabatra4** [reply]
Hi, yes it is possible to create a SQL binding using the v2 programming model. [Here](https://github.com/Azure/azure-functions-python-worker/blob/dev/tests/endtoend/sql_functions/sql_functions_stein/function_app.py) are examples of how you can use the SQL triggers and bindings.

**Reply by LouisDeconinck** [reply]
Thank you @shreyabatra4, that is definitely helpful. I managed to query my database, but when trying to add data to my Azure SQL database, I keep getting this error: 
`System.Private.CoreLib: Exception while executing function: Functions.AddRow. Microsoft.Azure.WebJobs.Host: Error while handling parameter _binder after function returned:. Microsoft.Azure.WebJobs.Extensions.Sql: Table dbo.InsertToDo does not exist.`

I am using this Azure function Python V2 code:
```Python
@app.function_name("AddRow")
@app.route(route="addrow")
@app.sql_output(arg_name="r",
                command_text="[dbo].[InsertToDo]",
                command_type="StoredProcedure",
                connection_string_setting="AzureWebJobsSqlConnectionString")
def addrow(req: func.HttpRequest, r: func.Out[func.SqlRow]) -> func.HttpResponse:
    try:
        # Pass the parameters to the stored procedure
        body = {"Id": 3, "title": "testtitle3"}

        row = func.SqlRow.from_dict(body)
        r.set(row)

        return func.HttpResponse(
            body=f"Inserted row with Id {body['Id']} and title {body['title']}",
            status_code=201,
            mimetype="text/plain"
        )
    except Exception as e:
        return func.HttpResponse(
            body=f"An error occurred: {str(e)}",
            status_code=500,
            mimetype="text/plain"
        )
```

As I managed to fetch data from the database, my connection string should be good.

I added this stored procedure:
```SQL
CREATE PROCEDURE InsertToDo
    @Id INT,
    @title NVARCHAR(50)
AS
BEGIN
    INSERT INTO dbo.ToDo (Id, title) VALUES (@Id, @title)
END
```

I tested this stored procedure in the Azure portal and it worked just fine.

Any idea what I might be doing wrong?

Is the recommended way to add new data through a stored procedure?
